lib/elixir/lib/process.ex

```diff
@@ -856,6 +860,10 @@ defmodule Process do
     end
   end
 
+  def info(pid, {:dictionary, key}) do
+    nilify(:erlang.process_info(pid, {:dictionary, key}))
+  end
+
   def info(pid, spec) when is_atom(spec) or is_list(spec) do
     nilify(:erlang.process_info(pid, spec))
```

Comment:
I would actually remove the specific clauses. Now that Erlang/OTP has good error messages, they will probably be more decent than function clause error?

```suggestion
  def info(pid, spec) do
    nilify(:erlang.process_info(pid, spec))
```

Reply:
On Erlang 26.2.2 the error looks like:
```
iex(1)> Process.info(self(), :unknown)
** (ArgumentError) errors were found at the given arguments:

  * 2nd argument: invalid item or item list

    :erlang.process_info(#PID<0.114.0>, :unknown)
    (elixir 1.17.0-dev) lib/process.ex:864: Process.info/2
    iex:1: (file)
```